src/sentry/integrations/vercel/uihook.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import
+
+import json
+
+from django.http import HttpResponse, HttpResponseNotFound
+from django.views.decorators.csrf import csrf_exempt
+
+from sentry.api.base import Endpoint
+from sentry.constants import ObjectStatus
+from sentry.models import Integration, Organization, OrganizationIntegration
+from sentry.web.api import allow_cors_options
+from sentry.web.helpers import render_to_response
+
+
+class VercelUIHook(Endpoint):
+    auth_required = False
+
+    @csrf_exempt
+    @allow_cors_options
+    def dispatch(self, request):
+        if request.method == "OPTIONS":
+            return HttpResponse(status=200)
+        if request.method != "POST":
+            return HttpResponseNotFound("Page not found")
+        body_unicode = request.body.decode("utf-8")
+        body = json.loads(body_unicode)
+        configuration_id = body["configurationId"]
+        user_id = body["user"]["id"]
+        team_id = body["teamId"]
+        integration = Integration.objects.get(external_id=team_id or user_id, provider="vercel")
+        if integration and integration.status == ObjectStatus.ACTIVE:
```

Comment:
@ceorourke The check for status should go in the query itself and the query should be wrapped in a `try/except` catching for the `Integration.DoesNotExist` error. Because as the code is currently written, if there isn't an `Integration` that matches the query conditions, it's going to blow up and return a 500 error.